recipes/data_bag.rb

```
@@ -0,0 +1,139 @@
+#
+# Cookbook Name:: osl-haproxy
+# Recipe:: shared
+#
+# Copyright 2015, Oregon State University
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+node.default['haproxy']['enable_ssl'] = false
+node.default['haproxy']['enable_default_http'] = false
+node.default['haproxy']['ssl_member_port'] = 443
+
+if node['osl-haproxy']['cluster'].nil?
+  Chef::Log.fatal("No cluster defined in node['osl-haproxy']['cluster']")
+  fail 'Error no cluster defined'
+end
+
+begin
+  config = data_bag("haproxy-#{node['osl-haproxy']['cluster']}")
+rescue
+  raise "Could not retrieve haproxy-#{node['osl-haproxy']['cluster']} data bag"
+end
+
+# set up global config hashes
+backend = Hash.new([])
+# store backend tmp so we can merge with member_#{foo} defaults.
+tmp_backends = Hash.new([])
+listeners = Hash.new([])
+frontend = Hash.new([])
+frontend['http'] = []
+frontend['https'] = []
+crt_names = []
+
+config.each do |project|
+  p_config = data_bag_item("haproxy-#{node['osl-haproxy']['cluster']}", project)
+
+  # Collect frontends
+  if p_config['frontend'].is_a? Enumerable
+    p_config['frontend'].each_pair do |listen, cfg|
+      if listen == 'https_http'
+        frontend['https'] += cfg
+        frontend['http'] += cfg
+      else
+        frontend[listen] += cfg
+      end
+    end
+  end
+
+  # Collect backends
+  if p_config['backend'].is_a? Enumerable
```

Sure. I'm curious why you're bothering to check if it's an `Enumerable`, when you don't really care if it isn't. It seems to me if it isn't, that it should throw a warn log. Otherwise it will fail to add the backend correctly, and fail silently.